spec/factories/submission.rb

```diff
@@ -2,5 +2,12 @@
   factory :submission do
     submission_type :population
     user
+
+    trait :finalized do
```

Comment:
Let's do it like that: I merge my PR and when merging yours, you will add what is important (mainly: setting submitted_object_id) from my trait to your (better) trait, and after all specs are ok, you'll remove my trait.